dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -79,9 +86,88 @@
+
+        return new Object[][] {
+                { container, templateContainer },
+                { fileContainer, templateFileContainer}
+        };
+    }
+
+    private static Template createTemplate(final Container container) throws DotSecurityException, WebAssetException, DotDataException {
+        final Template template = new TemplateDataGen().title("PageContextBuilderTemplate"+System.currentTimeMillis())
+                .host(site)
+                .withContainer(container, UUID).nextPersisted();
+        PublishFactory.publishAsset(template, systemUser, false, false);
+        return template;
+    }
+
+    private static Container createContainer() throws DotSecurityException, DotDataException, WebAssetException {
+        Container container = new Container();
+        final String containerName = "containerHTMLPageRenderedTest" + System.currentTimeMillis();
+
+        container.setFriendlyName(containerName);
+        container.setTitle(containerName);
+        container.setOwner(systemUser.getUserId());
+        container.setMaxContentlets(5);
+
+        final List<ContainerStructure> csList = new ArrayList<ContainerStructure>();
+        final ContainerStructure cs = new ContainerStructure();
+        cs.setStructureId(contentGenericType.id());
+        cs.setCode("$!{body}");
+        csList.add(cs);
+
+        container = APILocator.getContainerAPI().save(container, csList, site, systemUser, false);
+        PublishFactory.publishAsset(container, systemUser, false, false);
+
+        return container;
+    }
+
+    private static FileAssetContainer createFileContainer() throws DotSecurityException, DotDataException, WebAssetException {
+        return createFileContainer(site);
+    }
+
+    private static FileAssetContainer createFileContainer(final Host host)
+            throws DotSecurityException, DotDataException, WebAssetException {
+
+        final String containerName = "containerHTMLPageRenderedTest" + System.currentTimeMillis();
+        FileAssetContainer container = new ContainerAsFileDataGen()
+                .host(host)
+                .folderName(containerName)
+                .contentType(contentGenericType, "$!{body}")
+                .nextPersisted();
+
+        container = (FileAssetContainer) APILocator.getContainerAPI().find(container.getInode(), systemUser, true);
+
+        final Folder folder = APILocator.getFolderAPI().findFolderByPath(container.getPath(), host, systemUser, true);
+        final List<FileAsset> containerFiles =
+                APILocator.getFileAssetAPI().findFileAssetsByFolder(folder, systemUser, true);
+
+        for (final FileAsset containerFile : containerFiles) {
+            ContentletDataGen.publish(containerFile);
+        }
+
+        return container;
+    }
+
     @BeforeClass
     public static void prepare() throws Exception {
 
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    public static void prepareGlobalData() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5067852)